stablehlo/transforms/VhloToVersion.cpp

```diff
@@ -77,6 +78,69 @@ FailureOr<Version> validateTargetVersion(llvm::StringRef versionRef,
   return targetVersion;
 }
 
+template <typename AttrOrType>
+bool isDialect(AttrOrType at, llvm::StringRef dialectName) {
+  return (at.getDialect().getNamespace() == dialectName);
+}
+
+LogicalResult isLegalAttribute(Operation* op, NamedAttribute const& attr,
+                               Version targetVersion) {
+  // TODO: Remove once builtin types are forked.
+  if (isDialect(attr.getValue(), "builtin")) {
+    return success();
+  }
+
+  auto attrInterface = dyn_cast<VersionedAttrInterface>(attr.getValue());
+  if (attrInterface && isLegalVersionForTarget(attrInterface, targetVersion)) {
+    return success();
+  }
+
+  LLVM_DEBUG(llvm::dbgs() << "failed to legalize attribute " << attr.getName()
+                          << " to version " << targetVersion);
+  return failure();
+}
+
+LogicalResult isLegalType(Operation* op, Type const& type,
+                          Version const& targetVersion) {
+  // TODO: Remove once builtin types are forked.
+  if (isDialect(type, "builtin") || isDialect(type, "shape")) {
+    return success();
+  }
+
+  auto typeInterface = dyn_cast<VersionedTypeInterface>(type);
+  if (typeInterface && isLegalVersionForTarget(typeInterface, targetVersion)) {
+    return success();
+  }
+
+  LLVM_DEBUG(llvm::dbgs() << "failed to legalize type " << type
+                          << " to version " << targetVersion << '\n');
+  return failure();
+}
+
+bool isLegalOpInTargetVersion(Operation* op, Version const& targetVersion) {
```

Comment:
"isLegalOp" to be consistent with the two functions above? (Same for "isLegalVersionForTarget" => "isLegalVersion").